lib/OpenQA/Worker.pm

```diff
@@ -304,7 +304,7 @@ sub init {
         my ($working_dir) = grep { $_ && -d } @working_dirs;
         unless ($working_dir) {
             $_ and log_debug("Found possible working directory for $host: $_") for @working_dirs;
```

Comment:
geez, I don't know at all how this line can behave. Is the first `$_` referring to the inner $_ or the outer one?!?

Reply:
to the `@working_dirs` array

Reply:
That loop isn't that hard to read (for Perl code). At least I'd assume both `$_` are simply referring to the loop element and are therefore identical. But the code is definitely horrible.

Reply:
yeah I can read the code but looping over all working dirs and if the current working dir is not there then looping over all working dirs within another inner loop and the only thing we do is output a debug message and then go to the next entry is not my style :)